utils/src/main/scala/za/co/absa/enceladus/utils/general/ConfigReader.scala

```diff
@@ -31,4 +40,83 @@ object ConfigReader {
   def readStringConfig(path: String, default: String): String = {
     readStringConfigIfExist(path).getOrElse(default)
   }
+
+  /**
+   * Given a configuration returns a new configuration which has all sensitive keys redacted.
+   *
+   * @param keysToRedact A set of keys to be redacted.
+   */
+  def getRedactedConfig(keysToRedact: Set[String]): Config = {
+    def withAddedKey(accumulatedConfig: Config, key: String): Config = {
+      if (config.hasPath(key)) {
+        accumulatedConfig.withValue(key, ConfigValueFactory.fromAnyRef(redactedReplacement))
+      } else {
+        accumulatedConfig
+      }
+    }
+
+    val redactingConfig = keysToRedact.foldLeft(ConfigFactory.empty)(withAddedKey)
+
+    redactingConfig.withFallback(config)
+  }
+
+  /**
+   * Flattens TypeSafe config tree and returns the effective configuration
+   * while redacting sensitive keys.
+   *
+   * @param keysToRedact A set of keys for which should be redacted.
+   * @return the effective configuration as a map
+   */
+  def getFlatConfig(keysToRedact: Set[String] = Set()): Map[String, AnyRef] = {
+    def redact(key: String, value: AnyRef): AnyRef = {
```

Comment:
Why is there another logic to redact? Why don't we reuse what we already have (`getRedactedConfig`):

```scala
def getFlatConfig(keysToRedact: Set[String] = Set()): Map[String, AnyRef] = {
  val redactedConfig = getRedactedConfig(keysToRedact)
  redactedConfig.entrySet().asScala
}
```

... this also makes me think that this method would get so short (and there is no reusing it) that both these lines can just become part of `logEffectiveConfigProps`.

Reply:
Good spot! Indeed `getRedactedConfig` can be reused. The implementation of `getFlatConfig()` became slightly different since entries need to be explicitly converted to String->AnyRef.

Still prefer leaving this as a separate method since flattened config might make sense by its own. Will make this method do 1 thing though - flattering the config.